frontend/src/lib/utils/sns-aggregator-converters.utils.ts

```
@@ -0,0 +1,290 @@
+});
+
+const convertDerived = ({
+  sns_tokens_per_icp,
+  buyer_total_icp_e8s,
+  cf_participant_count,
+  direct_participant_count,
+  cf_neuron_count,
+}: CachedSnsSwapDerivedDto): SnsSwapDerivedState => ({
+  sns_tokens_per_icp,
+  buyer_total_icp_e8s: BigInt(buyer_total_icp_e8s),
+  cf_participant_count: nonNullish(cf_participant_count)
+    ? toNullable(BigInt(cf_participant_count))
+    : [],
+  direct_participant_count: nonNullish(direct_participant_count)
+    ? toNullable(BigInt(direct_participant_count))
+    : [],
+  cf_neuron_count: nonNullish(cf_neuron_count)
+    ? toNullable(BigInt(cf_neuron_count))
+    : [],
+});
+
+const convertDerivedToResponse = ({
+  sns_tokens_per_icp,
+  buyer_total_icp_e8s,
+  cf_participant_count,
+  direct_participant_count,
+  cf_neuron_count,
+}: CachedSnsSwapDerivedDto): SnsGetDerivedStateResponse => ({
+  sns_tokens_per_icp: toNullable(sns_tokens_per_icp),
+  buyer_total_icp_e8s: toNullable(
+    convertOptionalNumToBigInt(buyer_total_icp_e8s)
+  ),
+  cf_participant_count: nonNullish(cf_participant_count)
+    ? toNullable(BigInt(cf_participant_count))
+    : [],
+  direct_participant_count: nonNullish(direct_participant_count)
+    ? toNullable(BigInt(direct_participant_count))
+    : [],
+  cf_neuron_count: nonNullish(cf_neuron_count)
+    ? toNullable(BigInt(cf_neuron_count))
+    : [],
+});
+
+const convertIcrc1Metadata = (
+  icrc1Metadata: CachedSnsTokenMetadataDto
+): IcrcTokenMetadataResponse => {
+  return icrc1Metadata.map(([key, value]) => {
+    if ("Int" in value) {
+      return [key, { Int: BigInt(value.Int[0]) }];
+    }
+    if ("Nat" in value) {
+      return [key, { Nat: BigInt(value.Nat[0]) }];
+    }
+    if ("Text" in value) {
+      return [key, { Text: value.Text }];
+    }
+    if ("Nat64" in value) {
+      return [key, { Nat: BigInt(value.Nat64[0]) }];
+    }
+    // We don't need nor support Blob, Array and Map yet
+    return [key, { Text: "unsopported" }];
```

Can we throw an error here or would that be too little "support"?

---

I'd not throw here. Tokens might have more metadata than we use. Therefore, I'd rather not throw just because a token has more metadata than we use.

This will just map the unsupported metadata to this `Text` entry.